src/main/java/org/gridsuite/modification/server/modifications/PropertiesUtils.java

```diff
@@ -21,32 +20,71 @@ public static void applyProperties(Identifiable<?> identifiable, ReportNode subR
     }
 
     private static void applyProperty(Identifiable<?> identifiable, FreePropertyInfos prop, ReportNode subReportNode) {
+        ReportNodeAdder adder = subReportNode.newReportNode();
         if (prop.isDeletionMark()) {
             if (identifiable.removeProperty(prop.getName())) {
-                subReportNode.newReportNode()
-                    .withMessageTemplate("propertyDeleted", "    Property ${name} deleted")
-                    .withUntypedValue("name", prop.getName())
-                    .withSeverity(TypedValue.INFO_SEVERITY)
-                    .add();
+                deletePropertyReportNode(adder, prop);
+                adder.add();
             }
         } else {
             String oldValue = identifiable.setProperty(prop.getName(), prop.getValue());
             if (oldValue != null) { // update
-                subReportNode.newReportNode()
-                    .withMessageTemplate("propertyChanged", "    Property ${name} changed : ${from} -> ${to}")
-                    .withUntypedValue("name", prop.getName())
-                    .withUntypedValue("from", oldValue)
-                    .withUntypedValue("to", prop.getValue())
-                    .withSeverity(TypedValue.INFO_SEVERITY)
-                    .add();
+                modifyPropertyReportNode(adder, prop);
+                adder.add();
             } else { // insert
-                subReportNode.newReportNode()
-                    .withMessageTemplate("propertyAdded", "    Property ${name} added with value ${value}")
-                    .withUntypedValue("name", prop.getName())
-                    .withUntypedValue("value", prop.getValue())
-                    .withSeverity(TypedValue.INFO_SEVERITY)
-                    .add();
+                createPropertyReportNode(adder, prop);
+                adder.add();
             }
         }
     }
+
+    public static void reportPropertiesInfos(Identifiable<?> identifiable, List<FreePropertyInfos> modificationInfosProperties,
+                                             String propertiesLabelKey, ReportNode subReportNode) {
+        Optional<List<FreePropertyInfos>> properties = Optional.ofNullable(modificationInfosProperties);
+        List<ReportNode> reportNodes = new ArrayList<>();
+        if (properties.isPresent()) {
+            ReportNode propertiesReporter = subReportNode.newReportNode().withMessageTemplate(propertiesLabelKey, PROPERTIES).add();
+            properties.get().forEach(property -> reportNodes.add(applyPropertyReportNode(identifiable, property)));
+            ModificationUtils.getInstance().reportModifications(propertiesReporter, reportNodes,
```

Comment:
It's weird to have another sub report node with message="Properties" here. But we have this problem for every sub section. We should fix it everywhere in another PR

Reply:
yes I agree